exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/BatchUpsertStatement.kt

```diff
@@ -40,7 +39,20 @@ open class BatchUpsertStatement(
             }
             else -> dialect.functionProvider
         }
-        return functionProvider.upsert(table, arguments!!.first(), onUpdate, onUpdateExclude, null, transaction, keys = keys)
+        return functionProvider.upsert(table, arguments!!.first(), onUpdate, onUpdateExclude, where, transaction, keys = keys)
+    }
+
+    override fun arguments(): List<Iterable<Pair<IColumnType<*>, Any?>>> {
```

Comment:
Don't know is it worth it, but some duplication in `UpsertStatement` and `BatchUpsertStatemetn` could be reduced because the logic in `InsertStatement` is almost the same. At least, if I replace this method with

```kotlin
override fun arguments(): List<Iterable<Pair<IColumnType<*>, Any?>>> {
    val whereArgs = QueryBuilder(true).apply { where?.toQueryBuilder(this) }.args
    return super.arguments().map {
        it + whereArgs
    }
}
```

the new test is still green.